retrieval/target.go

```diff
@@ -196,11 +198,14 @@ func (t *Target) Update(cfg *config.ScrapeConfig, baseLabels, metaLabels clientm
 	if cfg.BasicAuth != nil {
 		t.url.User = url.UserPassword(cfg.BasicAuth.Username, cfg.BasicAuth.Password)
 	}
+	t.url.RawQuery = cfg.Params.Encode()
```

Comment:
How does this interact with metrics_path?

For SNMP, I want to put the address in URL sent to the exporter